mne/io/edf/edf.py

```diff
@@ -550,8 +550,14 @@ def _read_edf_header(fname, exclude):
         hour, minute, sec = [int(x) for x in
                              re.findall(r'(\d+)', fid.read(8).decode())]
         century = 2000 if year < 50 else 1900
-        meas_date = datetime(year + century, month, day, hour, minute, sec,
-                             tzinfo=timezone.utc)
+        try:
+            meas_date = datetime(year + century, month, day, hour, minute, sec,
+                                 tzinfo=timezone.utc)
+        except ValueError:
+            warn(f'Invalid date encountered ({year + century:04d}-{month:02d}-'
+                 f'{day:02d} {hour:02d}:{minute:02d}:{sec:02d}), setting it to'
+                 '1900-01-01 00:00:00.')
+            meas_date = datetime(1900, 1, 1, 0, 0, 0, tzinfo=timezone.utc)
```

Comment:
we should have a consistent default date across readers.

pb appeared in nirs format recently and anon has already 2 different defaults 

can you have a look?

Reply:
I hope that we're using DATE_NONE for this (so in this case you'd just set it to `None`)

Reply:
We're not. Should we? `DATE_NONE = (0, 2 ** 31 - 1)`, I'm not sure how the various readers or more importantly users would handle this value.

Currently, we have #8128 discussing an issue with anonymization where we have 1970-01-01, 2000-01-01, and the BIDS recommendation 1920-01-01. For NIRx, we have #8219 where invalid dates are set to 2000-01-01.

Reply:
>  I'm not sure how the various readers or more importantly users would handle this value.

Users would just see `info['meas_date'] = None`, and this is all you would do in this part of the code. The idea of `DATE_NONE` having these weird int values is just an I/O round-trip FIF detail.

Reply:
(to me this is probably the correct thing to use if we don't know the meas date, it's our default for example in `create_info`)

Reply:
Makes sense, I'll update the PR and we should probably do the same thing in other related PRs.

Reply:
Hi, I opened the #8281 datetime 'bug'. 
From user's side, `info['meas_date'] = None` makes sense and agreed that it probably is the most correct thing to output. I suggested the dummy date because I have seen similar corrections in other parts of the code. 
Thank you for working on this issue so fast, really great!

Reply:
Thanks @wolfgang-ganglberger, the test should now be fixed so this change will be in the latest dev version soon.